src/arrayMethodSort.js

```diff
@@ -4,8 +4,27 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let callBack = compareFunction;
+
+    if (!callBack) {
+      callBack = function (a, b) {
+        return String(a) > String(b) ? 1 : -1;
+      };
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (callBack(this[i], this[j]) > 0) {
+          const backup = this[i];
+
+          this[i] = this[j];
+          this[j] = backup;
+        }
+      }
+    }
```

Comment:
You're using the bubble sort algorithm here. While this is perfectly fine for this task, keep in mind that bubble sort is not very efficient for large arrays due to its quadratic time complexity. As a suggestion, you could look into more efficient sorting algorithms like quicksort or mergesort for future tasks.